R/deprecate.R

```diff
@@ -1,12 +1,52 @@
 #' These functions will be deprecated. Please use other functions instead.
 #' 
+#' @param x A
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#'   
+#' @param clust.col A single character value indicating the name of the 
+#'   \code{rowData} (or \code{colData}) where the data will be stored.
+#'   
+#' @param exprs_values a single \code{character} value for specifying which
+#'   assay to use for calculation.
+#'   (Please use \code{assay.type} instead.)
+#'   
+#' @param transposed Logical scalar, is x transposed with cells in rows?
+#' 
+#' @param type the type of measure used for the goodness of fit. One of
+#'   \sQuote{laplace}, \sQuote{AIC} or \sQuote{BIC}.
+#'   
+#' @inheritParams bluster::clusterRows
+#' @inheritParams runDMN
+#' @inheritParams transformAssay
+#' 
 #' @param x a \code{\link{SummarizedExperiment}} object -
 #' 
-#' @param ... -
+#' @param ... Additional parameters to use altExps for example
 #' 
 #' @name deprecate
 NULL
 
+#' @rdname deprecate
+setGeneric("cluster", signature = c("x"),
+            function(x, BLUSPARAM, assay.type = assay_name, 
+                    assay_name = "counts", MARGIN = "features", full = FALSE, 
+                    name = "clusters", clust.col = "clusters", ...)
+                standardGeneric("cluster"))
+
+#' @rdname deprecate
+#' @export
+#' @importFrom bluster clusterRows
+setMethod("cluster", signature = c(x = "SummarizedExperiment"),
+            function(x, BLUSPARAM, assay.type = assay_name, 
+                    assay_name = "counts", MARGIN = "features", full = FALSE, 
+                    name = "clusters", clust.col = "clusters", ...){
+                .Deprecated(msg = paste0("'cluster' is deprecated. ",
+                                        "Use 'addCluster' instead."))
+                addCluster(x)
```

Comment:
This does not pass parameters. addCluster gets only x parameter. Not others. Moreover, you can skip listing parameters. Use ... and then pass them to addCluster with ...


Only x is needed to list since the method is chosen based on class of x

Reply:
The idea is to keep these functions a hollow as possible. Just that they work but do not do anything but call the new function